src/ray/core_worker/core_worker.cc

```diff
@@ -849,8 +874,12 @@ CoreWorker::CoreWorker(CoreWorkerOptions options, const WorkerID &worker_id)
   // Unfortunately the raylet client has to be constructed after the receivers.
   if (task_receiver_ != nullptr) {
     task_argument_waiter_ = std::make_unique<DependencyWaiterImpl>(*local_raylet_client_);
+    // TODO(swang): Pass in fetch_callback. This should be a Python function
+    // that takes in the ObjectRefs plus RayActorObjectMetadata and calls the
+    // torch.distributed.recv.
+    p2p_task_argument_waiter_ = std::make_unique<P2pDependencyWaiter>(options_.fetch_p2p_dependency_callback);
```

Comment:
`p2p_task_argument_waiter_` is a wrapper of NCCL recv, and it is passed to `task_receiver_` and then passed to `actor_scheduling_queue`.